### PR TITLE
63-create-new-lineage-structure

### PR DESCRIPTION
**Pull request by ngaruko**
# Description

[description]

https://github.com/medic/medic-release-testing/issues/63

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by newtewt** [comment]
There is a bit of file structure change after my PR merges and I believe a couple of the changes you have here are included there. You'll need to rebase once that happens.

File: tests/page-objects/contacts/contacts.wdio.page.js
Comment (newtewt) [inline tests/page-objects/contacts/contacts.wdio.page.js:13]: I'm wondering if these elements need to go into a config specific page object. I don't know the html enough to say it's different between one config to the other. What are your thoughts?

File: tests/page-objects/contacts/contacts.wdio.page.js
Comment (newtewt) [inline tests/page-objects/contacts/contacts.wdio.page.js:54]: As a NIT, webdriver io has  `element.addValue()` function.  The difference being `setValue` clears the field first.

File: tests/page-objects/forms/generic-form.wdio.page.js
Comment (newtewt) [inline tests/page-objects/forms/generic-form.wdio.page.js:11]: sendKeys isn't available in wdio.

File: tests/page-objects/forms/generic-form.wdio.page.js
Comment (newtewt) [inline tests/page-objects/forms/generic-form.wdio.page.js:33]: There's a bit of duplication here. No need for the suffix of native anymore. After that there are two editForm functions. Can you remove the unnecessary one?

File: tests/page-objects/forms/generic-form.wdio.page.js
Comment (newtewt) [inline tests/page-objects/forms/generic-form.wdio.page.js:38]: Is this used? The chaining here won't work async.

File: tests/page-objects/forms/generic-form.wdio.page.js
Comment (newtewt) [inline tests/page-objects/forms/generic-form.wdio.page.js:1]: It seems like this page object was just pulled over and some portions were updated.  Can we remove what isn't being used to avoid having a mix of protractor syntax and wdio?

Comment (newtewt) [reply tests/page-objects/forms/generic-form.wdio.page.js:1]: It's also a good time to remove the native suffix as well.

**Review by newtewt — CHANGES_REQUESTED** [review]
nice! Looks good. Some questions and changes needed.

File: tests/page-objects/contacts/contacts.wdio.page.js
Comment (ngaruko) [reply tests/page-objects/contacts/contacts.wdio.page.js:13]: > I'm wondering if these elements need to go into a config specific page object. I don't know the html enough to say it's different between one config to the other. What are your thoughts?

I think these come from forms and they look same for both [default](https://github.com/medic/cht-core/blob/master/config/default/forms/contact/clinic-create.xml#L127) and [standrad](https://github.com/medic/cht-core/blob/master/config/standard/forms/contact/clinic-create.xml#L152)

File: tests/e2e/contacts/new-lineage-structure.wdio-spec.js
Comment (newtewt) [inline tests/e2e/contacts/new-lineage-structure.wdio-spec.js:20]: It would be real cool if we could find a wdio or mocha hook that does this for us and not have to add it everywhere.

**Review by newtewt — APPROVED** [review]
Nice work! Just one thing I missed on the original go through. Also, if you find a hook that runs after each file to reset the database so we can code it once that'd be awesome. Don't block this from merging to find that hook.

File: tests/page-objects/contacts/contacts.wdio.page.js
Comment (ngaruko) [reply tests/page-objects/contacts/contacts.wdio.page.js:77]: will fix this

File: tests/e2e/contacts/new-lineage-structure.wdio-spec.js
Comment (ngaruko) [reply tests/e2e/contacts/new-lineage-structure.wdio-spec.js:20]: will have a look at this as well. we also need a hook or hooks for before and/or beforeEach

**Pull request merged**
